ui/src/app/pages/homepage/recent-workspaces.tsx

```diff
@@ -42,35 +42,29 @@ export const RecentWorkspaces = withUserProfile()
 
   render() {
     // Needs a min-height so the spinner will render when loading and position: relative so said spinner will center.
-    return <React.Fragment>
-      {
-        this.state.loading ?
-          <SpinnerOverlay dark={true} /> :
-
-          this.state.recentWorkspaces.length === 0 ?
-
-            <div style={{color: colors.primary, margin: '.5em 2em'}}>
-              <h2 style={{fontWeight: 600, lineHeight: 1.5}}>Create your first Workspace</h2>
-              <div>As you create your workspaces, this area will store your most recent workspaces.</div>
-              <div>To see all workspaces created, click on <b>See all Workspaces</b> to the right.</div>
-            </div> :
-
-            <div>
-              <FlexRow style={{marginTop: '1rem', minHeight: 247, position: 'relative'}}>
-                {
-                  this.state.recentWorkspaces.map(recentWorkspace => {
-                    return <WorkspaceCard
-                      key={recentWorkspace.workspace.name}
-                      userEmail={this.props.profileState.profile.username}
-                      workspace={recentWorkspace.workspace}
-                      accessLevel={recentWorkspace.accessLevel}
-                      reload={() => this.loadWorkspaces()}
-                    />;
-                  })
-                }
-              </FlexRow>
-            </div>
-      }
-    </React.Fragment>;
+    return <div style={{position: 'relative'}}>
+      {this.state.loading && <SpinnerOverlay dark={true} />}
```

Comment:
Is this more useful than the original ternary because the page may have already loaded but the component may reload and we want the items in the component to be covered by the spinner?

Reply:
In my testing, I didn't see the dark overlay when the spinner was running. I thought it would have had a dark overlay on the recent workspaces section. 

<img width="1618" alt="Screen Shot 2019-11-12 at 2 52 34 PM" src="https://user-images.githubusercontent.com/5375000/68705492-30ad1100-055c-11ea-97cf-701621b4769d.png">

Reply:
Talked offline, this appears to be a pretty common thing.